### ISSUE TITLE
save_checkpoint 时有概率卡停整个训练过程

### ISSUE THREAD

**Issue by Ming-er** [SWivid/F5-TTS#691]
### Checks

- [X] This template is only for bug reports, usage problems go with 'Help Wanted'.
- [X] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [X] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### Environment Details

Ubuntu 22.04 LTS, RTX 4090D x 4, accelerate==1.2.1, torch==2.3.0+cu118, torchaudio==2.3.0+cu118, transformers                ==4.47.1

### Steps to Reproduce

accelerate launch --mixed_precision=fp16 src/f5_tts/train/train.py --config-name F5TTS_Small_train.yaml ++datasets.batch_size_per_gpu=19200

**以下是 yaml 文件的具体细节**
hydra:
  run:
    dir: /xxx/f5tts/1/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}/${now:%Y-%m-%d}/${now:%H-%M-%S}
    
datasets:
  name: LibriTTS_100_360_500
  batch_size_per_gpu: 38400  # 8 GPUs, 8 * 38400 = 307200
  batch_size_type: frame  # "frame" or "sample"
  max_samples: 64  # max sequences per batch if use frame-wise batch_size. we set 32 for small models, 64 for base models
  num_workers: 16

optim:
  epochs: 150
  learning_rate: 1e-4
  num_warmup_updates: 5000  # warmup steps
  grad_accumulation_steps: 2  # note: updates = steps / grad_accumulation_steps
  max_grad_norm: 1.0  # gradient clipping
  bnb_optimizer: False  # use bnb 8bit AdamW optimizer or not

model:
  name: F5TTS_Small
  tokenizer: char
  tokenizer_path: None  # if tokenizer = 'custom', define the path to the tokenizer you want to use (should be vocab.txt)
  arch:
    dim: 768
    depth: 18
    heads: 12
    ff_mult: 2
    text_dim: 512
    conv_layers: 4
    checkpoint_activations: False  # recompute activations and save memory for extra compute
  mel_spec:
    target_sample_rate: 24000
    n_mel_channels: 100
    hop_length: 256
    win_length: 1024
    n_fft: 1024
    mel_spec_type: vocos  # 'vocos' or 'bigvgan'
  vocoder:
    is_local: False  # use local offline ckpt or not
    local_path: None  # local vocoder path

ckpts:
  logger: tensorboard  # wandb | tensorboard | None
  save_per_updates: 20000  # save checkpoint per steps
  last_per_steps: 20000  # save last checkpoint per steps
  save_dir: /xxx/f5tts/1/${model.name}_${model.mel_spec.mel_spec_type}_${model.tokenizer}_${datasets.name}

 **以下是 accelerate_config 的具体细节，**
compute_environment: LOCAL_MACHINE
debug: false
distributed_type: MULTI_GPU
downcast_bf16: 'no'
enable_cpu_affinity: false
gpu_ids: all
machine_rank: 0
main_training_function: main
mixed_precision: fp16
num_machines: 1
num_processes: 4
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false

  

### ✔️ Expected Behavior

我在 libritts 上训练时，开始时设置 last_per_steps = 20000，结果模型在 20000 个 steps 后卡住了（显存没有超出，利用率为 100%，且我本人确定存储 ckpt 的目录还有很大存储空间），然后一段时间报错如下，
<img width="742" alt="image" src="https://github.com/user-attachments/assets/914969f3-6f88-415b-b23c-4d612abb27b4" />
我发现卡住的位置可能和  last_per_steps 有关，于是我将  last_per_steps 改为 1000，结果前两次 save checkpoint 是正常的，第三次又卡住了，
<img width="737" alt="image" src="https://github.com/user-attachments/assets/d3535fde-5d6e-4e73-af53-140d1774ea2c" />
此外，我的同门在使用 8 x 3090 复现时也遇到了类似的问题

**Comment by SWivid** [comment]
could try https://github.com/pytorch/pytorch/issues/111478

p.s. `using English to submit this report in order to facilitate communication.`

**Comment by wanghuii1** [comment]
If grad_accumulation_steps: 2 is set in the yaml, this issue will occur, but not with grad_accumulation_steps: 1. There might be a bug in trainer.py; it is recommended to fix it.
https://github.com/SWivid/F5-TTS/blob/4872afef9f81117f4d4b373dc98ed5ed09d77039/src/f5_tts/model/trainer.py#L361

> could try [pytorch/pytorch#111478](https://github.com/pytorch/pytorch/issues/111478)
> 
> p.s. `using English to submit this report in order to facilitate communication.`

**Comment by SWivid** [comment]
@wanghuii1 it's likely with `accelerate`
#346. the synchronization has some bug there

**Comment by Ming-er** [comment]
> could try [pytorch/pytorch#111478](https://github.com/pytorch/pytorch/issues/111478)
> 
> p.s. `using English to submit this report in order to facilitate communication.`

Unfortunately, It doesn't work.

**Comment by Ming-er** [comment]
Sorry to bother, are there any ideas to fix it? I cannot conduct the training due to the issue.

> @wanghuii1 it's likely with `accelerate` #346. the synchronization has some bug there

**Comment by SWivid** [comment]
@Ming-er sure, will add a pr for this see if will work

**Comment by SWivid** [comment]
@Ming-er feel free to check if #711 works. TIA~
